docs/guides/python/llama-rag.mdx

```diff
@@ -0,0 +1,309 @@
+
+  return ctx
+
+async def query_model(prompt: str):
+  print(f"Querying model: \"{prompt}\"")
+
+  # Get the model from the stored local context
+  if os.path.exists(persist_dir):
+    storage_context = StorageContext.from_defaults(persist_dir=persist_dir)
+
+    index = load_index_from_storage(storage_context)
+
+    # Get the query engine from the index, and use the prompt template for santisation.
+    query_engine = index.as_query_engine(
+      streaming=False,
+      similarity_top_k=4,
+      text_qa_template=text_qa_template
+    )
+  else:
+    print("model does not exist")
+    return "model does not exist"
+
+  # Query the model
+  query_response = query_engine.query(prompt)
+
+  print(f"Response: \n{query_response}")
+
+  return query_response.response
+
+Nitric.run()
+```
+
+## Test it locally
+
+Now that we have the Websocket defined, we can test it locally. You can do this using `nitric start` and connecting to the websocket through either the [Nitric Dashboard](/get-started/foundations/projects/local-development#local-dashboard) or another Websocket client. Once connected, you can send a message with a prompt to the model. Sending a prompt like "What is Nitric?" should produce an output similar to:
+
+```text
+Nitric is a cloud-agnostic framework designed to aid developers in building full cloud applications, including infrastructure.
+```
+
+## Get ready for deployment
+
+Now that its tested locally, we can get our project ready for containerization. The default python dockerfile uses `python3.11-bookworm-slim` as its basic container image, which doesn't have the right dependencies to load the Llama model. So, all we need to do is update the Dockerfile to use python3.11-bookworm (the non-slim version) instead.
+
+Update line 2:
+
+```dockerfile title:python.dockerfile
+# !diff -
+FROM ghcr.io/astral-sh/uv:python3.11-bookworm-slim AS builder
+# !diff +
+FROM ghcr.io/astral-sh/uv:python3.11-bookworm AS builder
+```
+
+And line 18:
+
+```dockerfile title:python.dockerfile
+# !diff -
+FROM python:3.11-slim-bookworm
+# !diff +
+FROM python:3.11-bookworm
+```
+
+When you're ready to deploy the project, we can create a new Nitric stack file that will target AWS:
+
+```bash
+nitric stack new dev aws
+```
+
+Update the stack file `nitric.dev.yaml` with the appropriate AWS region and memory allocation to handle the model:
+
+<Note>WebSockets are supported across all of AWS regions</Note>
+
+```yaml title:nitric.dev.yaml
+provider: nitric/aws@1.14.0
+region: us-east-1
+config:
+  # How services will be deployed by default, if you have other services not running models
+  # you can add them here too so they don't use the same configuration
+  default:
+    lambda:
+      # Set the memory to 6GB to handle the model, this automatically sets additional CPU allocation
+      memory: 6144
+      # Set a timeout of 30 seconds (this is the most API Gateway will wait for a response)
+      timeout: 30
```

Comment:
the same limit doesn't apply to websockets, they can run for up to 15 minutes now :)